python/ray/exceptions.py

```diff
@@ -130,8 +129,21 @@ def __init__(
         self.traceback_str = traceback_str
         self.actor_repr = actor_repr
         self._actor_id = actor_id
-        # TODO(edoakes): should we handle non-serializable exception objects?
         self.cause = cause
+
+        try:
+            pickle.dumps(cause)
+        except (pickle.PicklingError, TypeError) as e:
+            err_msg = (
+                "The original cause of the RayTaskError"
+                f" ({self.cause.__class__}) isn't serializable: {e}."
+                " Overwriting the cause to a RayError."
+            )
+            logger.warning(err_msg)
+            self.cause = RayError(err_msg)
+
+        self.args = (function_name, traceback_str, self.cause, proctitle, pid, ip)
```

Comment:
Could you move the comments (line 119-121) to here as well since they are comments for self.args?

Reply:
fixed